daml-lf/scenario-interpreter/src/main/scala/com/digitalasset/daml/lf/ScenarioRunner.scala

```diff
@@ -421,20 +427,31 @@ object ScenarioRunner {
         case SResultError(err) =>
           SubmissionError(Error.RunnerException(err), enrich(onLedger.incompleteTransaction))
         case SResultNeedContract(coid, committers, callback) =>
-          ledger.lookupContract(coid, committers, readAs, callback) match {
-            case Left(err) => SubmissionError(err, enrich(onLedger.incompleteTransaction))
-            case Right(_) => go()
+          // TODO (drsk) Reduce duplication between this code and the engine. 
+          // https://github.com/digital-asset/daml/issues/14090
+          discTable.contractById.get(coid) match {
+            case None =>
+              ledger.lookupContract(coid, committers, readAs, callback) match {
+                case Left(err) => SubmissionError(err, enrich(onLedger.incompleteTransaction))
+                case Right(_) => go()
+              }
+            case Some(_) => go()
```

Comment:
I'm not familiar with this code, but I don't see how this provides the contract needed by `SResultNeedContract`. Perhaps this should use the `callback` like the other branch?